osquery/events/events.cpp

```diff
@@ -61,11 +61,10 @@ static inline EventTime timeFromRecord(const std::string& record) {
 
 static inline std::string toIndex(size_t i) {
   auto j = std::to_string(i);
-  size_t n = 1;
-  while (i /= 10) {
-    n++;
+  if (j.size() < 10) {
+    j = std::string(10 - j.size(), '0').append(std::move(j));
   }
-  return (n >= 10) ? j : std::string(10 - n, '0').append(std::move(j));
+  return j;
```

Comment:
You could use `stl` instruments, it is more clear, I think:
```c++
std::ostringstream ostr;
ostr << std::setfill('0') << std::setw(10) << i;
return ostr.str();
```

Reply:
I'm afraid it's slower, is not it?

Reply:
I'd say `stl` should be faster. Your version consist at least 1 more coping of number-in-string (`j`) and at least 2 memory allocations. Stl, I think, should do everything in-place with 1 memory alllocation.

But I didn't make any measurements.

Reply:
Yes, you are right, `stringstream` is a slow.

Reply:
You also could use `std::string::insert`. It must be even faster.
```c++
j.insert(j.begin(), 10 - j.size(), '0');
```
And may be it would be good idea to change a name of index string representation from `j` to something more readable (`str`, `stringIndex` or whatever you like).